### DISCUSSION TITLE
能否增加自定义sql脚本提交时的校验规则

### DISCUSSION THREAD

**Discussion by JaySem** [hhyo/Archery#1855] [Ideas]
针对SQL提交时的SQL脚本检测有2个疑问：1  SQL检测规则调用了哪些内部文件?  2 如果想要增加一些SQL规则(比如create 表时，超过50个字段就报错)，进行二次开发，需要修改哪些内部文件?

**Comment by nick2wang** [comment]
mysql的sql检测是调用[goinception](https://github.com/hanchuanchuan/goInception)服务，可以对goinception进行二开；其他数据库的检测在sql/engines/对应的db类型中的execute_check方法里实现

**Comment by JaySem** [comment]
谢谢

**Comment by JaySem** [comment]
请问，v1.8.5版本的archery, 对应的goinception代码是在哪?
